pkg/pool-utils/contracts/factories/BasePoolSplitCodeFactory.sol

```diff
@@ -36,17 +37,20 @@ import "@balancer-labs/v2-solidity-utils/contracts/helpers/SingletonAuthenticati
  * prevent the creation of any future pools from the factory.
  */
 abstract contract BasePoolSplitCodeFactory is IBasePoolSplitCodeFactory, BaseSplitCodeFactory, SingletonAuthentication {
+    IProtocolFeePercentagesProvider private _protocolFeeProvider;
```

Comment:
```suggestion
    IProtocolFeePercentagesProvider private immutable _protocolFeeProvider;
```

Reply:
I wondered why it wasn't immutable in the other contracts